src/base/misc/dos2linux.c

```diff
@@ -615,6 +615,7 @@ uint64_t do_read_qword(dosaddr_t addr, sim_pagefault_handler_t handler)
 static int check_write_pagefault(dosaddr_t addr, uint32_t op, int len,
 				 sim_pagefault_handler_t handler)
 {
+  e_invalidate(addr, len);
```

Comment:
Maybe move that to the callers?
`check_write_pagefault()` just doesn't suggest
(by its name) to be the right place for things
that should work no matter whether the
write should page-faulted.

Also while looking at the callers, I've found
this code:
```
void do_write_dword(dosaddr_t addr, uint32_t dword, sim_pagefault_handler_t hand
ler)
{
  if (mem_likely_protected(addr, 4)) {
    if (((addr+3) & (PAGE_SIZE-1)) < 3) {
      do_write_word(addr, dword & 0xffff, handler);
      do_write_word(addr+2, dword >> 16, handler);
    }
```
Isn't a `return` missing after splitting the write?

Reply:
all done now